src/panels/config/entities/entity-registry-settings.ts

```diff
@@ -1136,24 +1164,41 @@ export class EntityRegistrySettings extends SubscribeMixin(LitElement) {
       this._submitting = false;
     }
 
-    if (this._switchAs !== "switch") {
+    if (this._switchAs !== computeDomain(this._entityId)) {
       if (
         !(await showConfirmationDialog(this, {
           text: this.hass!.localize(
-            "ui.dialogs.entity_registry.editor.switch_as_x_confirm",
+            `ui.dialogs.entity_registry.editor.${
+              this._helperConfigEntry?.domain === "switch_as_x"
+                ? "switch_as_x_remove_confirm"
+                : "switch_as_x_confirm"
+            }`,
             "domain",
-            this._switchAs
+            domainToName(
+              this.hass.localize,
+              this._helperConfigEntry?.domain === "switch_as_x"
+                ? domain
+                : this._switchAs
+            ).toLowerCase()
           ),
         }))
       ) {
         return;
       }
+      const entityId = this._entityId.trim();
+      if (this._helperConfigEntry?.domain === "switch_as_x") {
+        // remove current helper
+        await deleteConfigEntry(this.hass, this._helperConfigEntry.entry_id);
+        if (this._switchAs === "switch") {
+          return;
+        }
+      }
```

Comment:
Can we move all these new logic into a dedicated function to avoid using many time the `this._helperConfigEntry?.domain === "switch_as_x"` condition?

```ts
if (this._helperConfigEntry?.domain === "switch_as_x") {
    this.removeSwitchAsXHelper()
    return;
}
```